impl/src/main/java/org/jboss/weld/bootstrap/FastAnnotatedTypeLoader.java

```diff
@@ -62,6 +64,7 @@
     public <T> SlimAnnotatedTypeContext<T> loadAnnotatedType(String className, String bdaId) {
         try {
             final ClassFileInfo classFileInfo = classFileServices.getClassFileInfo(className);
+            final boolean  checkTypeModifiers = isJandexSetFlagsAvailable();
```

Comment:
I believe it will be more efficient to declare this on the `FastAnnotatedTypeLoader` class. The same applies to `BootstrapLogger.LOG.usingOldJandexVersion()`.